File: phase 1/01 - two pointers/lc26.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)

        if n<2:
            return n

        lp = 0
        rp = 1

        while rp<n:
            if nums[lp] != nums[rp]:
                lp+=1
                nums[lp] = nums[rp]
            rp+=1
        
        return lp+1

File: phase 1/01 - two pointers/lc26.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    
    let n = nums.length
    if (n<2) {return n}

    let lp = 0
    let rp = 1

    while(rp < n){
        if(nums[lp]==nums[rp]){
            rp++
        } else{
            nums[lp+1] = nums[rp]
            rp++
            lp++
        }

    }

    return lp+1

};